### PR TITLE
Enhance JSONRecordReader to handle GZIP compressed JSON files.

### PR DESCRIPTION
**Pull request by mayankshriv**
Enhanced JSONRecordReader to handle reading of GZIP compressed JSON files.
Used existing way in `RecordReaderUtils` to check for GZIP compression that
just relies on `.gz` segment generation.

TODO:
Consider identifying GZIP compression based on `GZIPInputStream.GZIP_MAGIC` instead
of file extension.

## Description
Add a description of your PR here.
A good description should include pointers to an issue or design document, etc.
## Upgrade Notes
Does this PR prevent a zero down-time upgrade? (Assume upgrade order: Controller, Broker, Server, Minion)
* [ ] Yes (Please label as **<code>backward-incompat</code>**, and complete the section below on Release Notes)

Does this PR fix a zero-downtime upgrade introduced earlier?
* [ ] Yes (Please label this as **<code>backward-incompat</code>**, and complete the section below on Release Notes)

Does this PR otherwise need attention when creating release notes? Things to consider:
- New configuration options
- Deprecation of configurations
- Signature changes to public methods/interfaces
- New plugins added or old plugins removed
* [ ] Yes (Please label this PR as **<code>release-notes</code>** and complete the section on Release Notes)
## Release Notes
If you have tagged this as either backward-incompat or release-notes,
you MUST add text here that you would like to see appear in release notes of the
next release.

If you have a series of commits adding or enabling a feature, then
add this section only in final commit that marks the feature completed.
Refer to earlier release notes to see examples of text

## Documentation
If you have introduced a new feature or configuration, please add it to the documentation as well.
See https://docs.pinot.apache.org/developers/developers-and-contributors/update-document

### CODE REVIEW

File: pinot-plugins/pinot-input-format/pinot-json/src/main/java/org/apache/pinot/plugin/inputformat/json/JSONRecordReader.java [hunk 1]
```diff
@@ -86,12 +91,14 @@ public GenericRow next(GenericRow reuse) {
   public void rewind()
       throws IOException {
     _iterator.close();
+    _inputStream.close();
```
Comment (fx19880617) [inline pinot-plugins/pinot-input-format/pinot-json/src/main/java/org/apache/pinot/plugin/inputformat/json/JSONRecordReader.java:94]: do you need to rewind the stream?

Comment (mayankshriv) [reply pinot-plugins/pinot-input-format/pinot-json/src/main/java/org/apache/pinot/plugin/inputformat/json/JSONRecordReader.java:96]: The `init()` should now take care of re-initializing the stream.

**Comment by fx19880617** [comment]
overall lgtm, can we add a test for the gzipped json file?

**Review by fx19880617 — APPROVED** [review]

**Comment by kishoreg** [comment]
how about other file formats. can we make sure we gzip for other formats as well?

**Comment by mayankshriv** [comment]
> how about other file formats. can we make sure we gzip for other formats as well?

It seemed like other file formats are already using the right input stream that supports reading GZIPed files, JSON record reader for some reason wasnt.

**Pull request merged**